auth/credentials/oauth.go

```diff
@@ -31,11 +39,33 @@ type OAuthTokenSource struct {
 	clientSecret string
 	userAgent    string
 	tokenURL     string
+	revokeURL    string
 	token        *Token
 	tokenCache   LocalTokenCache
 	ctx          context.Context
 }
 
+func (c *OAuthTokenSource) RevokeToken() error {
+	tokenString, err := c.tokenCache.RetrieveToken(c.ctx)
+	if err != nil {
+		return err
+	}
+	if tokenString != nil && *tokenString != "" {
```

Comment:
i guess string instead of *string would also help here

Reply:
Exactly. I will do dedicated PR for this change to make sure that we do not have any leftovers in the docs etc.